rust_src/src/cmds.rs

```diff
@@ -1,18 +1,70 @@
+//! Simple built-in editing commands.
+
 use lisp::LispObject;
 use lisp::defsubr;
 use remacs_macros::lisp_fn;
-use remacs_sys::EmacsInt;
+use remacs_sys::{set_point, EmacsInt, Fline_end_position};
 use threads::ThreadState;
 
+
 /// Return buffer position N characters after (before if N negative) point.
 #[lisp_fn]
 pub fn forward_point(n: LispObject) -> LispObject {
     let pt = ThreadState::current_buffer().pt();
     LispObject::from_fixnum(n.as_fixnum_or_error() + pt as EmacsInt)
 }
 
+/// Move point to end of current line (in the logical order).
+/// With argument N not nil or 1, move forward N - 1 lines first.
+/// If point reaches the beginning or end of buffer, it stops there.
+/// To ignore intangibility, bind `inhibit-point-motion-hooks' to t.
+///
+/// This function constrains point to the current field unless this moves
+/// point to a different line than the original, unconstrained result.  If
+/// N is nil or 1, and a rear-sticky field ends at point, the point does
+/// not move.  To ignore field boundaries bind `inhibit-field-text-motion'
+/// to t.
+#[lisp_fn(min = "0", intspec = "^p")]
+pub fn end_of_line(n: LispObject) -> LispObject {
+    let mut num = if n.is_nil() {
+        LispObject::from_fixnum(1)
+    } else {
+        n
+    };
+    num.as_fixnum_or_error();
+    let mut newpos: isize;
+    let mut pt: isize;
+    let cur_buf = ThreadState::current_buffer();
+    loop {
+        newpos = unsafe {
+            LispObject::from(Fline_end_position(num.to_raw())).as_fixnum_or_error() as isize
+        };
+        unsafe { set_point(newpos) };
+        pt = cur_buf.pt();
+        if pt > newpos as isize && cur_buf.fetch_char(pt) == '\n' as i32 {
```

Comment:
`newpos` is already isize?

Also, `fetch_char(pt)` vs `FETCH_CHAR(pt-1)` in the original.